testsuite/python/oif_volume_conservation.py

```diff
@@ -94,16 +91,42 @@ def check_relaxation(self, **kwargs):
         self.assertGreater(prefactor, 0.)
         self.assertAlmostEqual(diameter_final, diameter_init, delta=0.005)
         self.assertAlmostEqual(lam, 0.0325, delta=0.0001)
+        self.system.thermostat.turn_off()
+        self.system.part.clear()
 
-    def test(self):
+    def test_00_volume_relaxation(self):
+        """
+        Load a soft elastic sphere via ``object_in_fluid``, stretch it and
+        check restoration of original volume due to elastic forces.
+        """
         self.assertEqual(self.system.max_oif_objects, 0)
-        with self.subTest(msg='linear stretching'):
+        with self.subTest(msg="linear stretching"):
             self.check_relaxation(kslin=1.)
         self.assertEqual(self.system.max_oif_objects, 1)
-        with self.subTest(msg='neo-Hookean stretching'):
+        with self.subTest(msg="neo-Hookean stretching"):
             self.check_relaxation(ks=1.)
         self.assertEqual(self.system.max_oif_objects, 2)
 
+    def test_01_elastic_forces(self):
+        half_box_l = np.copy(self.system.box_l) / 2.
+        cell = oif.OifCell(cell_type=self.make_cell_type(),
+                           particle_type=0, origin=half_box_l)
+        # stretch cell
+        partcls = self.system.part.all()
+        partcls.pos = (partcls.pos - half_box_l) * 1.5 + half_box_l
+        # reduce number of triangles to speed up calculation
+        cell.mesh.triangles = cell.mesh.triangles[:20]
+        # smoke test
+        results = cell.elastic_forces(el_forces=6 * [0], f_metric=6 * [1])
+        ref = [0., 1.36730815e-12, 22.4985704, 6838.5749, 7.3767594, 6816.6342]
```

Comment:
where do these nice numbers come from?

Reply:
It's a verbatim copy of the output of the elastic forces calculation. I get the same values on ESPResSo 4.2.1, so the main branch is consistent with the last release. The second value in the list is sensitive to numerical errors and gives different results on ARM vs x86 architectures, which is why a non-zero absolute tolerance is used. As far as I understand it, these numbers should be fully deterministic on a given hardware.

Reply:
not the best physics test then but the best we get I guess